src/packageManager/downloadAndInstallPackages.ts

```diff
@@ -61,6 +61,11 @@ export async function downloadAndInstallPackages(
                 eventStream.post(new InstallationFailure(installationStage, error));
             }
 
+            // If the package is optional, log and continue with the next package
+            if (pkg.isOptional) {
+                continue;
```

Comment:
we should log telemetry on the failure (probably including the exact failure kind)